node/src/validator/router.rs

```diff
@@ -38,6 +46,77 @@ impl<N: Network, C: ConsensusStorage<N>> P2P for Validator<N, C> {
     }
 }
 
+impl<N: Network, C: ConsensusStorage<N>> Validator<N, C> {
+    /// An extended handshake dedicated to consensus committee members.
+    async fn committee_handshake(
+        &self,
+        peer_ip: SocketAddr,
+        mut framed: Framed<&mut TcpStream, MessageCodec<N>>,
+    ) -> io::Result<()> {
+        // Establish quorum with other validators:
+        //
+        // 1. Sign and send the node's pub key.
+        // 2. Receive and verify peer's signed pub key.
+        // 3. Insert into connected_committee_members.
+        // 4. If quorum threshold is reached, start the bft.
+
+        // 1.
+        // BFT must be set here.
+        // TODO: we should probably use something else than the public key, potentially interactive, since this could
+        // be copied and reused by a malicious validator.
+        let public_key = self.primary_keypair.public();
+        let signature = self
+            .primary_keypair
+            .private()
+            .sign_bytes(public_key.to_bytes().as_slice(), &mut rand::thread_rng())
```

Comment:
Seems we should be signing the `last_executed_sub_dag_index`, otherwise run the risk of a MITM attack as well.